src/plan_expand_hypertable.c

```diff
@@ -25,22 +25,85 @@
 #include "hypertable.h"
 #include "hypertable_restrict_info.h"
 #include "planner_import.h"
-
-
+#include "extension.h"
+#include "chunk.h"
+#include "catalog/pg_type.h"
+#include "utils/errcodes.h"
 
 typedef struct CollectQualCtx
 {
 	PlannerInfo *root;
 	RelOptInfo *rel;
-	List	   *result;
+	List	   *restrictions;
+	FuncExpr   *chunk_exclusion_func;
 } CollectQualCtx;
 
+static Oid	chunk_exclusion_func = InvalidOid;
+#define CHUNK_EXCL_FUNC_NAME "chunks_in"
+static Oid	ts_chunks_arg_types[] = {RECORDOID, INT4ARRAYOID};
+
+static void
+init_chunk_exclusion_func()
+{
+	if (chunk_exclusion_func == InvalidOid)
+		chunk_exclusion_func = get_function_oid(CHUNK_EXCL_FUNC_NAME, ts_extension_schema_name(), lengthof(ts_chunks_arg_types), ts_chunks_arg_types);
+}
 
 static bool
+is_chunk_exclusion_func(Node *node)
+{
+	if (IsA(node, FuncExpr))
+	{
+		FuncExpr   *explicit_exclusion = (FuncExpr *) node;
+
+		if (explicit_exclusion->funcid == chunk_exclusion_func)
+			return true;
+	}
+	return false;
+}
+
+static bool
+contains_chunk_exclusion_func(Node *node)
```

Comment:
Should be removed now since unhandled invocations will error, also this doesnt detect all nested uses.

Reply:
I think it's better to catch things early on, even if it does not cover all the possible problems.